core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -467,6 +468,9 @@ public Frame decorrelateRel(LogicalAggregate rel) {
     }
     final RelNode newInput = frame.r;
 
+    // aggregate outputs mappings: group keys + aggregates
+    Map<Integer, Integer> outputsMapping = Maps.newHashMap();
```

Comment:
make final; use `new HashMap<>()`; rename to outputMap; can you re-use `combinedMap`?

Reply:
OK;

`combinedMap` could not be re-used because `combinedMap` could only be used for `newProject`, not for `postProject`.